tensorboard/data/server/data_compat.rs

```diff
@@ -214,6 +262,180 @@ mod tests {
+                pb::TensorProto {
+                    dtype: pb::DataType::DtFloat.into(),
+                    tensor_shape: Some(tensor_shape(&[])),
+                    float_val: vec![0.125, 9.99],
+                    ..Default::default()
+                },
+                pb::TensorProto {
+                    dtype: pb::DataType::DtDouble.into(),
+                    tensor_shape: Some(tensor_shape(&[])),
+                    double_val: vec![0.125, 9.99],
+                    ..Default::default()
+                },
+                pb::TensorProto {
+                    dtype: pb::DataType::DtFloat.into(),
+                    tensor_shape: Some(tensor_shape(&[])),
+                    tensor_content: [f32::to_le_bytes(0.125), f32::to_le_bytes(9.99)]
+                        .iter()
+                        .flatten()
+                        .copied()
+                        .collect(),
+                    ..Default::default()
+                },
+                pb::TensorProto {
+                    dtype: pb::DataType::DtDouble.into(),
+                    tensor_shape: Some(tensor_shape(&[])),
+                    tensor_content: [f64::to_le_bytes(0.125), f64::to_le_bytes(9.99)]
+                        .iter()
+                        .flatten()
+                        .copied()
+                        .collect(),
+                    ..Default::default()
+                },
+            ];
+            for tensor in tensors {
+                let v = EventValue::Summary(SummaryValue(Box::new(Value::Tensor(tensor.clone()))));
+                let expected = Ok(ScalarValue(0.125));
+                let actual = v.into_scalar();
+                assert_eq!(
+                    actual, expected,
+                    "into_scalar for {:?}: got {:?}, expected {:?}",
+                    &tensor, actual, expected
+                )
+            }
+        }
+
+        #[test]
+        fn test_enrich_non_float_tensors() {
+            let tensors = vec![
+                pb::TensorProto {
+                    dtype: pb::DataType::DtString.into(),
+                    string_val: vec![b"abc".to_vec()],
+                    ..Default::default()
+                },
+                pb::TensorProto {
+                    dtype: pb::DataType::DtInt32.into(),
+                    int_val: vec![123],
+                    ..Default::default()
+                },
+                pb::TensorProto::default(),
+            ];
+            for tensor in tensors {
+                let v = EventValue::Summary(SummaryValue(Box::new(Value::Tensor(tensor.clone()))));
+                let expected = Err(DataLoss);
+                let actual = v.into_scalar();
+                assert_eq!(
+                    actual, expected,
+                    "into_scalar for {:?}: got {:?}, expected {:?}",
+                    &tensor, actual, expected
+                )
+            }
+        }
+
+        #[test]
+        fn test_enrich_graph_def() {
```

Comment:
Yeah, but these are all in the `tests::scalars` module, so I think it’s
fine. Part of the reason that I made this a module was so that the names
wouldn’t have to be so long and redundant.

(For that matter, I could be convinced to leave the `test_` prefix off
the function names, since it’s signaled out of band with `#[test]`. I’ve
kept it just out of muscle memory, really, from \*Unit. A quick grep of
rustc shows that about half their `#[test]`s have `test_`.)